app/publish/tus.go

```diff
@@ -0,0 +1,204 @@
+package publish
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/lbryio/lbrytv/app/auth"
+	"github.com/lbryio/lbrytv/app/proxy"
+	"github.com/lbryio/lbrytv/app/query/cache"
+	"github.com/lbryio/lbrytv/app/rpcerrors"
+	"github.com/lbryio/lbrytv/app/sdkrouter"
+	"github.com/lbryio/lbrytv/internal/errors"
+	"github.com/lbryio/lbrytv/internal/metrics"
+	"github.com/lbryio/lbrytv/internal/monitor"
+	"github.com/lbryio/lbrytv/internal/responses"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	tusd "github.com/tus/tusd/pkg/handler"
+	"github.com/ybbus/jsonrpc"
+)
+
+const module = "tus.publish"
+
+// PublishHandler handle media publishing on odysee-api, it implements TUS
+// specifications to support resumable file upload and extends the handler to
+// support fetching media from remote url.
+type PublishHandler struct {
+	*tusd.UnroutedHandler
+
+	logger       monitor.ModuleLogger
+	composer     *tusd.StoreComposer
+	authProvider auth.Provider
+}
+
+// NewHandler creates a new publish handler.
+func NewHandler(authProvider auth.Provider, cfg tusd.Config, uploadPath string) (*PublishHandler, error) {
```

Comment:
An unwritten convention is that `NewSomething` should return `Something`, not `SomeSomething`. But again, I think the handler should just be renamed.